flask_restful/fields.py

```diff
@@ -415,3 +421,24 @@ def _iso8601(dt):
     :return: A ISO 8601 formatted date string
     """
     return dt.isoformat()
+
+
+def _rfc3339(dt):
+    """Turn a datetime object into an RFC3339 formatted date at UTC.
+
+    Example::
+
+        fields._rfc3339(datetime(2012, 1, 1, 0, 0, tzinfo=pytz.utc)) => "2012-01-01T00:00:00Z"
+
+    :param dt: The datetime to transform
+    :type dt: datetime
+    :return: A RFC 3339 formatted date string
+    """
+    if dt.tzinfo is None or dt.tzinfo.utcoffset(dt) is None:
+        raise MarshallingException(
+            'Naive datetime %s not supported by RFC 3339.' % dt
+        )
+    d = dt.astimezone(pytz.utc)
+    ts = timegm(d.timetuple())
```

Comment:
That's why it's important to use `astimezone()` with UTC, because you will lose the offset if you do `timetuple()` on a non-UTC datetime.

```python
import pytz
import datetime
from calendar import timegm

dt1 = datetime.datetime.now(tz=pytz.timezone('CET'))
dt2 = dt1.astimezone(pytz.utc)

print("dt1", dt1)
print("dt2", dt2)

ts1 = timegm(dt1.timetuple())
ts2 = timegm(dt2.timetuple())

print("dt1", datetime.datetime.fromtimestamp(ts1, tz=pytz.utc))
print("dt2", datetime.datetime.fromtimestamp(ts2, tz=pytz.utc))
```

```
dt1 2015-08-19 01:14:38.786696+02:00
dt2 2015-08-18 23:14:38.786696+00:00
dt1 2015-08-19 01:14:38+00:00  # WRONG
dt2 2015-08-18 23:14:38+00:00
```

(Microseconds are lost in this example, but the `_rfc3339` function handles them below this line.)